component/src/main/java/org/ballerinalang/kafka/util/KafkaUtils.java

```diff
@@ -513,4 +533,20 @@ private static void addBooleanParamIfPresent(String paramName,
         return partitionList;
     }
 
+    public static List<BMap<String, BValue>> createPartitionList(Context context,
+                                                                   Collection<TopicPartition> partitions) {
+
+        List<BMap<String, BValue>> topicPartitionList = new ArrayList<>();
+        if (!partitions.isEmpty()) {
```

Comment:
If partition list is empty what would be the implication ?

Reply:
Then empty list will be returned.